backend/controller/dal/dal.go

```diff
@@ -752,6 +752,9 @@ func (d *DAL) ReplaceDeployment(ctx context.Context, newDeploymentKey model.Depl
 	var replacedDeploymentKey optional.Option[model.DeploymentKey]
 	oldDeployment, err := tx.GetExistingDeploymentForModule(ctx, newDeployment.ModuleName)
 	if err == nil {
+		if oldDeployment.Key.String() == newDeploymentKey.String() {
```

Comment:
Are you sure this is a good idea? We used to do this and it resulted in a lot of errors being spewed out due to rebuilds occurring but the deployments being identical. Maybe things have changed though, this was a while ago.